### PR TITLE
Fix: LockManagerTest.updateValue is flaky

### PR DESCRIPTION
**Pull request by shibd**
### Motivation

#13663 Flaky-test: org.apache.pulsar.metadata.LockManagerTest.updateValue

The reasons are discussed in detail in this PR(#13725)

This is mainly caused by the callback method of `MetadataStore` and the concurrent access of the notification thread to the `MetadataCache#refresh`.

We should queue them in the `metada-store` thread pool(SingleThread)

### Modifications
 1. In `AbstractMetadataStore` method, all future callback opt are executed through `metada-store` thread.
 2. Fix unit `testSharedInstance` fail.
 Two MetadataStore do not share a listener, when executed store2.delete, store1 existsCache is not invalidate.
https://github.com/apache/pulsar/blob/11bfc0e95dc210c0dce70d1f53eb6eb43dbe835d/pulsar-metadata/src/test/java/org/apache/pulsar/metadata/impl/LocalMemoryMetadataStoreTest.java#L51-L62

### Documentation
- [ x] `no-need-doc`

### CODE REVIEW

**Comment by shibd** [comment]
@Jason918 @Technoboy- @codelipenghui Hi, Can you help me review it again? Thanks.

**Comment by nodece** [comment]
This PR looks the same as #13809, we can find this solution together.

**Comment by shibd** [comment]
> This PR looks the same as #13809, we can find this solution together.

Thank you reply, I saw you fix the callback problem of ZKMetadata. I'll cancel this commit. 

However, the essential reason why this unit test will report an error is `LocalMemoryMetadataStore` that  does not callback through `metada-store` thread pool.

I think it can be handled like this: 
>In AbstractMetadataStore method, all future callback opt are executed through metada-store thread.

**Comment by nodece** [comment]
@shibd You are right, I don't notice the memory  metadata store.

File: pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/AbstractMetadataStore.java [hunk 1]
```diff
@@ -227,15 +227,15 @@ public void accept(Notification n) {
         }
         // Ensure caches are invalidated before the operation is confirmed
         return storeDelete(path, expectedVersion)
-                .thenRun(() -> {
+                .thenRunAsync(() -> {
```
Comment (Jason918) [inline pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/AbstractMetadataStore.java:230]: `storeDelete` should ensure that the callback is called in executor?

Comment (codelipenghui) [reply pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/AbstractMetadataStore.java:230]: +1, is it necessary to use `thenRunAsync`?

Comment (shibd) [reply pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/AbstractMetadataStore.java:230]: @Jason918  @codelipenghui Sorry for the late response. I think we should ensure, In this way, the cache can be prevented from being executed concurrently

Comment (nodece) [reply pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/AbstractMetadataStore.java:230]: You cannot use the executor to execute this, this callback thread is sometimes executed in the executor thread: https://github.com/apache/pulsar/blob/master/pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/ZKMetadataStore.java#L330

Comment (Jason918) [reply pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/AbstractMetadataStore.java:230]: > In this way, the cache can be prevented from being executed concurrently

I think it's better to ensure that in `storeDelete`, to reduce this extra context switch.

**Comment by shibd** [comment]
@Jason918  @codelipenghui  @nodece Sorry,The current changes have caused other problems. I will resubmit the PR in another way

File: pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/LocalMemoryMetadataStore.java [hunk 2]
```diff
@@ -84,6 +88,14 @@ public LocalMemoryMetadataStore(String metadataURL, MetadataStoreConfig metadata
             // Use a reference from a shared data set
             String name = uri.getHost();
             map = STATIC_MAPS.computeIfAbsent(name, __ -> new TreeMap<>());
+            STATIC_INSTANCE.compute(name, (key, value) -> {
```
[the remark this answers is not in the document]
Comment (Jason918) [reply pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/LocalMemoryMetadataStore.java:91]: > The main repair here is that the following unit tests will fail.

Is this in the scope of this PR?  If not, it's better to open anther PR.

File: pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/AbstractMetadataStore.java (hunk 1 above)
Comment (shibd) [reply pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/AbstractMetadataStore.java:230]: > You cannot use the executor to execute this, this callback thread is sometimes executed in the executor thread: https://github.com/apache/pulsar/blob/master/pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/ZKMetadataStore.java#L330

There seems to be this problem.

File: pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/LocalMemoryMetadataStore.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (shibd) [reply pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/LocalMemoryMetadataStore.java:91]: @Jason918 Thank you reply, I will open ISSUE to ##13908

File: pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/AbstractMetadataStore.java (hunk 1 above)
Comment (shibd) [reply pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/AbstractMetadataStore.java:230]: @nodece @Jason918 @codelipenghui I resubmitted PR #13911 , please have a look, thank you.

**Pull request closed without merging**